api/v1alpha1/openapi.yaml

```diff
@@ -1307,6 +1307,390 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /api/v1/certificatesigningrequests:
+    get:
+      tags:
+        - certificatesigningrequest
+      description: list CertificateSigningRequests
+      operationId: listCertificateSigningRequests
+      # We can add further parameters as needed based on
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1589
+      # Others available are allowWatchBookmarks, fieldSelector, resourceVersion, resourceVersionMatch, sendInitialEvents, timeoutSeconds, watch
+      parameters:
+        - name: continue
+          in: query
+          description: An optional parameter to query more results from the server. The value of the paramter must match the value of the 'continue' field in the previous list response.
+          required: false
+          schema:
+            type: string
+        - name: labelSelector
+          in: query
+          description: A selector to restrict the list of returned objects by their labels. Defaults to everything.
+          schema:
+            type: string
+        - name: limit
+          in: query
+          description: The maximum number of results returned in the list response. The server will set the 'continue' field in the list response if more results exist. The continue value may then be specified as parameter in a subsequent query.
+          required: false
+          schema:
+            type: integer
+            format: int32
+      responses:
+        "200":
+        # TODO - k8s has a bunch of other options for the content, do we want these? some seem k8s-specific
+        # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1686
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequestList'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+    # TODO: add parameters? k8s has dryRun, fieldManager, fieldValidation
+    # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1743
+      tags:
+        - certificatesigningrequest
+      description: request Certificate Signing
+      operationId: createCertificateSigningRequest
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CertificateSigningRequest'
+        required: true
+      responses:
+      # TODO - do we want "208" already reported, "400" bad request, "409" status conflict? Enrollment Request has these
```

Comment:
We return 400 on input validation errors. 208 sounds reasonable if we want CSRs to be immutable. I see we're not using 409 on the ERs, hm....

Reply:
We can always add 409 later as needed.